pennylane/ops/qubit/arithmetic_ops.py

```diff
@@ -305,3 +310,257 @@ def compute_decomposition(wires):
+                    ]
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+                    for control_values in control_values_list:
+                        mat = mat @ MultiControlledX.compute_matrix(
+                            control_wires, control_values=control_values
+                        )
+
+            else:
+                if value == 0:
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+
+                else:
+                    control_values_list = [format(n, binary) for n in range(value)]
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+                    for control_values in control_values_list:
+                        mat = mat @ MultiControlledX.compute_matrix(
+                            control_wires, control_values=control_values
+                        )
+
+        else:
+            raise ValueError(f"The compared value must be an int. Got {type(value)}.")
+
+        return mat
+
+    @staticmethod
+    def compute_decomposition(value, geq=True, wires=None, **kwargs):
+        r"""Representation of the operator as a product of other operators (static method).
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        .. seealso:: :meth:`~.IntegerComparator.decomposition`.
+
+        Args:
+            value (int): The value :math:`L` that the state's decimal representation is compared against.
+            geq (bool): If set to ``True``, the comparison made will be :math:`\sigma \geq L`. If ``False``, the comparison
+                made will be :math:`\sigma < L`.
+            wires (Union[Wires, Sequence[int], or int]): Control wire(s) followed by a single target wire where
+                the operation acts on.
+
+        Returns:
+            list[Operator]: decomposition into lower level operations
+
+        **Example:**
+
+        >>> print(qml.IntegerComparator.compute_decomposition(4, wires=[0, 1, 2, 3]))
+        [MultiControlledX(wires=[0, 1, 2, 3], control_values="100"),
+         MultiControlledX(wires=[0, 1, 2, 3], control_values="101"),
+         MultiControlledX(wires=[0, 1, 2, 3], control_values="110"),
+         MultiControlledX(wires=[0, 1, 2, 3], control_values="111")]
+        """
+
+        if not isinstance(value, int):
+            raise ValueError(f"The compared value must be an int. Got {type(value)}.")
+        if wires is None:
+            raise ValueError("Must specify the wires that the operation acts on.")
+        if len(wires) > 1:
+            control_wires = Wires(wires[:-1])
+            wires = Wires(wires[-1])
+        else:
+            raise ValueError(
+                f"IntegerComparator: wrong number of wires. {len(wires)} wire(s) given. Need at least 2."
+            )
+
+        work_wires = Wires("aux") if len(control_wires) > 2 else None
+        binary = "0" + str(len(control_wires)) + "b"
+
+        if geq:
+            if value > 2 ** len(control_wires) - 1:
+                gates = [Identity(0)]
```

Comment:
Similar minor suggestions as per refactoring the logic here slightly as for the matrix computation above.